tests/devices/qubit/test_simulate.py

```diff
@@ -16,10 +16,11 @@
 import numpy as np
 import pytest
 from dummy_debugger import Debugger
+from scipy.stats import ttest_1samp
```

Comment:
Is this the right test to use here? Should we consider other suitable tests?

Reply:
For this test since we're generating the samples ourselves and verifying that the sample distribution is as we expect, the ttest seems suitable. @mudit2812 what do you think?

Reply:
Yeah I was under the same impression as @astralcai . What makes you consider other tests @mlxd ?

Reply:
I am wondering if a https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.kstest.html is a better fit here:

_"two-sided: The null hypothesis is that the two distributions are identical, F(x)=G(x) for all x; the alternative is that they are not identical."_

We aim to compare the results for being the same, which matches with our intention, right?

Reply:
I think that both tests are viable. The `kstest` is comparing two distributions, while the `ttest` is comparing a distribution against a known mean value. The latter seems easier to implement as a unit test, as for a toy circuit, we can numerically compute the expectation value and compare a set of samples against that.

Meanwhile, I'm struggling to think of a way we can generate an "expected" sample distribution in a nice way. Either we use numerically computed probabilities to compute samples with numpy, or we use an "expected circuit" which uses `defer_measurements` and returns samples. The latter is a pretty weak unit testing strategy.

Reply:
Hi team. I come from Pennylane Lightning and we have a similar problem. We have a test for different measurements with different observables. However, we set the number of shots around 1_000_000 to be sure to get closer values to the reference. The problem with this approach is the time-consuming for the test. Therefore, I am trying to find an alternative using statistical analysis to optimize the shot number. Currently, I am developing this [proposal](https://github.com/PennyLaneAI/pennylane-lightning/pull/881/files).

As you mentioned here, I was trying to use `ttest_1samp`. Unfortunately, this test applies just if the data has a normal distribution.  I found a different test called `kruskal` which is independent of the distribution and handles multiple samples. 

I am not an expert on statistics but my research approach me to this solution. 

Do you think that these Hypothesis Tests could help you?

Reply:
My main concern with `t-test` is that the data to analyze should have a normal distribution. I am not sure if the samples of this test will have a normal distribution. Otherwise, you might consider using non-parametric alternatives such as Kruskal. :sweat_smile:

Reply:
@maliasadi